### ISSUE TITLE
use JustJ to add JREs to the product?

### ISSUE THREAD

**Issue by chirontt** [howlger/IDEalized#4]
The `post-processing` code in `product/pom.xml`, to add a JRE (only for Windows platform) to the final product:
https://github.com/howlger/IDEalized/blob/6c4ce7dad2ca480334c42e9f86bead805021d64f/product/pom.xml#L71-L74
is a bit messy and clumsy.

To make it simpler, I suggest to use the [JustJ](https://eclipse.dev/justj/) feature to add JREs to all supported platforms (i.e. not just Windows). The description in the JustJ page is a bit confusing and too much, but its application is straightforward.

I may create a PR to do this, if it's accepted.

**Comment by howlger** [comment]
Yes, the embedding of an [IBM Semeru JDK](https://developer.ibm.com/semeru-runtime-downloads/) via `jre` subfolder could be better. Embedding a Temurin JDK via JustJ as an updatable feature is much simpler:

- In the [product](https://github.com/howlger/IDEalized/blob/master/product/idealized.product#L33-L105) add the feature `org.eclipse.justj.openjdk.hotspot.jre.full`
- In the [target platform](https://github.com/howlger/IDEalized/blob/master/target-platform/idealized-target.target) add a location like:
    ```xml
    <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="false" type="InstallableUnit">
        <unit id="org.eclipse.justj.openjdk.hotspot.jre.full.feature.group" version="0.0.0"/>
        <repository location="https://download.eclipse.org/justj/jres/21/updates/release"/>
    </location>
    ```
- Disable (or remove) the embedding of a JDK via `jre` subfolder in [`/product/pom.xml`](https://github.com/howlger/IDEalized/blob/master/product/pom.xml) by setting the [property `add-jre`](https://github.com/howlger/IDEalized/blob/6c4ce7dad2ca480334c42e9f86bead805021d64f/product/pom.xml#L13) to `no`

I personally prefer to use OpenJ9 over the HotSpot Java VM, as the startup time and resource consumption is to my experience significantly lower. Maybe an update site could be built in a separate project or in an preceding step similar to `mirror-non-p2`, from which the JDK can then be embedded as simple as with JustJ (JustJ could be the default and would be replaced by running this step upfront). A pull request for this would be very welcome. What do you think?

**Comment by chirontt** [comment]
I have no experience with the IBM Semeru JDK, so I can't comment on it. But this issue is about the JustJ JREs which are exclusively OpenJDK/Temurin, so the IBM JDK consideration is out of scope for this issue.

I don't understand what the `mirror-non-p2` stuff is all about, so I can't comment on it either.

Also, I don't think providing an update site for the embedded JRE is a good idea, as updating it (the embedded JRE) would require a new product release/version, I assume, i.e. updating the embedded JREs would go together with releasing a new product version.

As this discussion is getting beyond my original idea (i.e. simplifying the embedding of the JREs with JustJ) for this issue, I think I will just close this issue without any PR, as a better issue should replace it.
